### PR TITLE
Add support for defining new interfaces and other minor things

### PR DESCRIPTION
**Pull request by sdroege**
See individual commits

### CODE REVIEW

File: src/subclass/interface.rs [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+                *(self as *mut _ as *mut ffi::GType),
+                name,
+                arg_types,
+                ret_type,
+                accumulator,
+            );
+        }
+    }
+
+    /// Add a new action signal with accumulator to the interface.
```
Comment (EPashkin) [inline src/subclass/interface.rs]: With accumulator?

**Comment by EPashkin** [comment]
Thanks, found only small nit in docs

File: src/subclass/interface.rs [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+    /// should be combined with the old value in the first argument. If no further
+    /// signal handlers should be called, `false` should be returned.
+    fn add_signal_with_accumulator<F>(
+        &mut self,
+        name: &str,
+        arg_types: &[Type],
+        ret_type: Type,
+        accumulator: F,
+    ) where
+        F: Fn(&mut Value, &Value) -> bool + Send + Sync + 'static,
```
Comment (EPashkin) [inline src/subclass/interface.rs]: Maybe use enum { None, TrueHandled, FirstWins, .., Custom(F:Fn())} for accumulator?
Not sure how to handle `g_application_handle_local_options_accumulator` etc,
maybe define trait to get function pointer?

File: src/subclass/interface.rs (hunk 1 above)
Comment (sdroege) [reply src/subclass/interface.rs]: What do you mean? :)

Also this is the same documentation as we already had for subclasses before

File: src/subclass/interface.rs (hunk 2 above)
Comment (sdroege) [reply src/subclass/interface.rs]: An enum would have the same problem that we had with `Option` for optional callbacks, and `Fn` is a trait for giving a function pointer and passing some data around ;)

What exactly do you see wrong with the way how the function is defined? Also this is how we had it before for subclasses already, no change here.

If you'd like to have a different API for that, maybe let's handle that in a separate PR?

Comment (sdroege) [reply src/subclass/interface.rs]: `g_application_handle_local_options_accumulator` would be

```rust
|return_acc, handler_return| {
  let value = handler_return.get::<u32>().unwrap();
  *return_acc = value.to_value();
  value < 0
}
```

File: src/subclass/interface.rs (hunk 1 above)
Comment (EPashkin) [reply src/subclass/interface.rs]: I missed that `handler` actually `GSignalAccumulator accumulator`, sorry

File: src/subclass/interface.rs (hunk 2 above)
Comment (EPashkin) [reply src/subclass/interface.rs]: Yes can handled later, just note about extension

**Comment by EPashkin** [comment]
👍

**Comment by sdroege** [comment]
Ok, let's get this in then? :) @GuillaumeGomez @EPashkin?

I added another small commit on top.

**Comment by EPashkin** [comment]
👍

**Comment by GuillaumeGomez** [comment]
All good for me!

**Comment by EPashkin** [comment]
Still 👍
Thanks @sdroege

**Pull request merged**